grails-app/controllers/org/pih/warehouse/product/ProductController.groovy

```diff
@@ -266,8 +275,12 @@ class ProductController {
                 // Need to validate here FIRST otherwise we'll run into an uncaught transient property exception
                 // when the session is closed.
                 if (productInstance.validate()) {
-                    if (!productInstance.productCode) {
-                        productInstance.productCode = productService.generateProductIdentifier(productInstance.productType)
+                    try {
+                        if (!productInstance.productCode) {
+                            productInstance.productCode = productService.generateProductIdentifier(productInstance.productType)
+                        }
+                    } catch (Exception e) {
+                        productInstance.errors.rejectValue("productCode", e.message)
```

Comment:
I'd for now let the errors handler do the job, instead of having to write a lot of `try/catches`. If the users complain about the "generic" error message for this one (I assume they wouldn't, because this error should be rare), we can refactor it in MVP1/2.
As refactor I'd rather mean refactor the errors handler.

Reply:
Sure I see your point and it makes sense so, for now, I'll just leave the error handling to the generic controller event handler but I am not too sure how are we going to handle these kinds of exceptions for template views generically, I still think we will have to manually try/catch but I guess we'll see on MVP2